build.zig

```diff
@@ -404,3 +406,110 @@ fn link_tracer_backend(
         },
     }
 }
+
+fn go_client(
+    b: *std.build.Builder,
+    header_generate_step: *std.build.Step,
+    mode: Mode,
+    options: *std.build.OptionsStep,
+    tracer_backend: TracerBackend,
+) void {
+    const build_step = b.step("go_client", "Build Go client shared library");
+
+    // Updates the generated header file:
+    const install_header = b.addInstallFile(
+        .{ .path = "src/clients/c/tb_client.h" },
+        "../src/clients/go/pkg/native/tb_client.h",
+    );
+
+    build_step.dependOn(header_generate_step);
+    build_step.dependOn(&install_header.step);
+
+    // Zig cross-targets
+    const platforms = .{
+        "x86_64-linux",
+        "x86_64-macos",
+        "x86_64-windows",
+        "aarch64-linux",
+        "aarch64-macos",
+    };
+
+    inline for (platforms) |platform| {
+        const cross_target = CrossTarget.parse(.{ .arch_os_abi = platform, .cpu_features = "baseline" }) catch unreachable;
+
+        const lib = b.addStaticLibrary("tb_client", "src/clients/c/tb_client.zig");
+        lib.setMainPkgPath("src");
+        lib.setTarget(cross_target);
+        lib.setBuildMode(mode);
+        lib.pie = true;
+        lib.bundle_compiler_rt = true;
+
+        if (cross_target.os_tag.? != .windows) {
+            lib.linkLibC();
+        }
```

Comment:
We can always linkLibC() on windows now (`builtin.link_libc` is used to [know when to export](https://github.com/tigerbeetledb/tigerbeetle/blob/main/src/c/tb_client.zig#L57-L65) the tb_client functions)

Reply:
Thanks for the tip! That's why the Java CI is failing right now on Windows!